client/src/modules/stock/import/stockImport.js

```diff
@@ -0,0 +1,108 @@
+angular.module('bhima.controllers')
+  .controller('StockImportController', StockImportController);
+
+// dependencies injections
+StockImportController.$inject = [
+  'DepotService', 'InventoryService', 'NotifyService', 'SessionService', 'util',
+  'bhConstants', 'ReceiptModal', 'StockFormService', 'StockService',
+  'StockModalService', 'uiGridConstants', 'appcache', 'Upload', '$state',
+];
+
+/**
+ * @class StockImportController
+ *
+ * @description
+ * This module helps to import stock from a file
+ */
+function StockImportController(
+  Depots, Inventory, Notify, Session, util, bhConstants, ReceiptModal, StockForm,
+  Stock, StockModal, uiGridConstants, AppCache, Upload, $state
+) {
+  const vm = this;
+
+  const cache = new AppCache('StockCache');
+
+  vm.changeDepot = changeDepot;
+  vm.downloadTemplate = Stock.downloadTemplate;
+
+  vm.submit = () => {
+    // send data only when a file is selected
+    if (!vm.depot.uuid || !vm.file) {
+      vm.noSelectedFile = true;
+      return null;
+    }
+
+    return uploadFile(vm.file);
+  };
+
+  /** upload the file to server */
+  function uploadFile(file) {
+    vm.uploadState = 'uploading';
+
+    const parameters = {
+      url : '/stock/import/',
+      data : { file, depot_uuid : vm.depot.uuid },
+    };
+
+    const filters = [
```

Comment:
Could you move these `filters` to outside the function?  Since they never change, there is no reason to have them in the `uploadFile()`.